waku/v2/utils/peers.nim

```diff
@@ -44,22 +44,23 @@ proc init*(p: typedesc[RemotePeerInfo],
 
   return remotePeerInfo
 
-proc initAddress(T: type MultiAddress, str: string): T {.raises: [Defect, ValueError, LPError].}=
-  # @TODO: Rather than raising exceptions, this should return a Result
-  let address = MultiAddress.init(str).tryGet()
-  if IPFS.match(address) and matchPartial(multiaddress.TCP, address):
-    return address
+## Check if wire Address is supported
+proc validWireAddr*(ma: MultiAddress): bool =
+  const
+    ValidTransports = mapOr(TCP, WebSockets)
+  if ValidTransports.match(ma) == true:
+    return true
   else:
-    raise newException(ValueError,
-                       "Invalid bootstrap node multi-address")
+    return false
```

Comment:
You can replace this whole conditional with:
```
return ValidTransports.match(ma)
```
This is simpler, since you are testing a boolean condition and returning the result of that condition.